File: Medium/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        initial_area = min(height[left], height[right]) * (right - left)
        
        while left != right:
            if height[left] < height[right]:
                left = left + 1
            else:
                right = right - 1
            
            new_area = min(height[left], height[right]) * (right - left)
            
            if new_area > initial_area:
                initial_area = new_area

        return initial_area

File: Medium/ContainerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int area = std::min(height[left], height[right]) * (right - left);

        while ( left != right ) {
            if ( height[left] < height[right] )
                left++;
            else
                right--;
            int new_area = std::min(height[left], height[right]) * (right - left);
            if ( new_area > area )
                area = new_area;
        }
        return area;
    }
};
